src/app/multiselect/multiselect.directive.js

```diff
@@ -100,14 +106,26 @@
                         }
                     }
 
-                    self.options.push({
+                    var optionObj = {
                         id: index,
                         label: multiselect.getLabel(option),
                         value: value,
                         selected: selected
-                    });
+                    };
+
+                    if (self.isGrouped) {
+                        optionObj.group = multiselect.getGroup(option);
+                    } else {
+                        optionObj.group = 'ungrouped';
+                    }
+
+                    _groupsHash[optionObj.group] = true;
+                    self.options.push(optionObj);
                 });
 
+                // Set the groups array
+                self.groups = Object.keys(_groupsHash);
```

Comment:
Good point. I am going to assume we should retain the order of the groups as they come up in the options.